common/enum_base.h

```diff
@@ -51,6 +58,44 @@ namespace Carbon::Internal {
 //   #include ".../my_kind.def"
 //   };
 //   ```
+//
+// The result of the above:
+// - An enum class ("RawEnumType") defined in an `Internal` namespace with one
+//   enumerator per call to CARBON_MY_KIND(Name) in ".../my_kind.def", with name
+//   `Name`.
+// - A type `MyKind` with a RawEnumType field that includes:
+//   - all the public members of `EnumBase`, like `name` and `Print`
```

Comment:
These were not sentences before, since they generally didn't have subject or verbs, but a list of things included in `MyKind`. I've rearranged to make these sentences.